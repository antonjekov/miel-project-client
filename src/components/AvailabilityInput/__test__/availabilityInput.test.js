import React from 'react'
import renderer from "react-test-renderer"
import {render, cleanup, screen, fireEvent} from "@testing-library/react"
import AvilabilityInput from '../'

describe('AvailabilityInpuet', ()=>{

    afterEach(cleanup)
    it('matches snapshot without error', () => {
        const values = {availability: 'Available'}
        const errors = {}
        const tree = renderer.create( <AvilabilityInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })
    
    it('matches snapshot with error', () => {
        const values = {availability: 'Available'}
        const errors = {availability: 'Some error'}
        const tree = renderer.create( <AvilabilityInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })

})
