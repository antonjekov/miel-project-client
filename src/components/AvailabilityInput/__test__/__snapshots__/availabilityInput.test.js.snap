// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailabilityInpuet matches snapshot with error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Availability
  </label>
  <select
    className="form-control form-control is-invalid"
    name="availability"
    onChange={[Function]}
    value="Available"
  >
    <option />
    <option>
      Available
    </option>
    <option>
      Not available
    </option>
  </select>
  <div
    className="invalid-feedback"
  >
    Some error
  </div>
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;

exports[`AvailabilityInpuet matches snapshot without error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Availability
  </label>
  <select
    className="form-control form-control is-valid"
    name="availability"
    onChange={[Function]}
    value="Available"
  >
    <option />
    <option>
      Available
    </option>
    <option>
      Not available
    </option>
  </select>
  <div
    className="invalid-feedback"
  />
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;
